### PR TITLE
hydrojudge&ui&core: allow superuser to sync testdata manually

### PR DESCRIPTION
**Pull request by StupidQu**
allow superuser to sync testdata manually with a longer time limit.

### CODE REVIEW

**Comment by pandadtdyy** [comment]
I advise you save your data on a s3 storage. Sync testdata is a invalid operation.

File: packages/ui-default/templates/partials/problem_sidebar_normal.html [hunk 1]
```diff
@@ -88,6 +88,11 @@ <h1 class="section__title">{{ pdoc['title'] }}</h1>
           <span class="icon icon-settings"></span> {{ _('Judge Config') }}
         </a></li>
       {% endif %}
+      {% if handler.user.hasPerm(perm.PERM_ALL) %}
+        <li class="menu__item"><a class="menu__link" href="{{ pdoc.docId }}/syncdata">
+          <span class="icon icon-copy"></span> {{ _('Sync TestData') }}
```
Comment (undefined-moe) [inline packages/ui-default/templates/partials/problem_sidebar_normal.html:93]: The user should acknowledge that it is NOT always required to sync testdata manually.

**Review by undefined-moe — COMMENTED** [review]
In my opinion, testdata sync should be a client pull operation instead of a server-push operation.

**Comment by StupidQu** [comment]
It also helps when you want to sync testdata before a contest to avoid waiting during the contest.

**Comment by undefined-moe** [comment]
PTAL at the review above and fix ci errors.

**Pull request closed without merging**